### ISSUE TITLE
Unable to build the angular files

### ISSUE THREAD

**Issue by skerit** [louis-lau/DuckyPanel#7]
Running the build command gives me this error:

```
An unhandled exception occurred: Cannot find module '@angular-devkit/build-angular/package.json'
Require stack:
- /usr/local/lib/node_modules/@angular/cli/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /usr/local/lib/node_modules/@angular/cli/node_modules/@angular-devkit/architect/node/index.js
- /usr/local/lib/node_modules/@angular/cli/models/architect-command.js
- /usr/local/lib/node_modules/@angular/cli/commands/build-impl.js
- /usr/local/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /usr/local/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/index.js
- /usr/local/lib/node_modules/@angular/cli/utilities/json-schema.js
- /usr/local/lib/node_modules/@angular/cli/models/command-runner.js
- /usr/local/lib/node_modules/@angular/cli/lib/cli/index.js
- /usr/local/lib/node_modules/@angular/cli/lib/init.js
- /usr/local/lib/node_modules/@angular/cli/bin/ng
See "/tmp/ng-TfZb6i/angular-errors.log" for further details.
```

**Comment by louis-lau** [comment]
Heya, I can't seem to reproduces this on my local setup. What's your node.js version? I'm currently on the recommended version 12.16.3.

**Comment by skerit** [comment]
I'm also on 12.16.3! But I was missing the `ng` command.
I accidentally installed `angular-cli` when I should have installed `@angular/cli`. Even though I did eventually do that, it broke something ...

However, now I have a typescript error:

    ERROR in The Angular Compiler requires TypeScript >=3.6.4 and <3.9.0 but 3.9.2 was found instead.

**Comment by louis-lau** [comment]
You should be able to use `npm run ng` which will use the angular cli package from the local dependencies. The locally installed typescript is also 3.8.3 instead of 3.9.2.

**Comment by skerit** [comment]
Ah, the version you defined in the package.json is: `^3.8.3`, with a caret. That'll allow npm to upgrade the patch AND the minor version, so any 3.x version greater than 3.8 is also allowed to install. You should change it to `~3.8.3` in order to limit it to `3.8.x` versions.

**Comment by louis-lau** [comment]
That would be the case without a package-lock.json. That makes sure the right versions get installed. If you take a look at the package-lock.json you'll see TypeScript 3.8.3

**Comment by skerit** [comment]
Ah, good point. Unfortunately `npm install` ignores the package-lock.json file. In order to get it to work, the `npm ci` command has to be used. (Well, that used to be true anyways. They've changed it so many times now, I haven't touched package-lock for years)

**Comment by louis-lau** [comment]
As far as I know that is not the case: https://docs.npmjs.com/configuring-npm/package-lock-json.html

> It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.

**Comment by louis-lau** [comment]
By the way, I just found the trouble you were having with unauthorized messages in DuckyPanel. A dependency update is to blame again. The unauthorized response from DuckyApi is now slightly different meaning Duckypanel doesn't catch it and redirect to login. Fix incoming 😁

**Comment by skerit** [comment]
Great! Is that why I keep getting `You do not have the permission to access this resource.` messages, even after I log into DuckyPanel with my DuckyAPI credentials?

Btw: I never immediately get the login screen. Whenever I log onto DuckyPanel, I see the regular interface. when I go to /profile, the top message says `You are logged in as , you've got access to NaN undefined of storage.`
It's only after I click `Logout` that I actually get the login screen :)

**Comment by louis-lau** [comment]
The 403 is probably because you're trying to login with your admin account. The admin account is only meant for user management. After creating a user using the admin account you can login as that user. Endpoints meant for the admin user are marked with [Admin only] in the docs.

The login screen problem is because of the changed unauthorized exception, on it! 😁

**Comment by louis-lau** [comment]
Should be fixed if you now pull the latest changes from both DuckyApi and DuckyPanel.

**Comment by louis-lau** [comment]
>The admin account is only meant for user management

Added a note to the readme to clear this up

**Comment by skerit** [comment]
Yup, everything seems to be working now!

Quick question: I already added an address to Wildduck before, is there a way to get those to show up in the panel?

**Comment by louis-lau** [comment]
Yes, this is something I should document. I answered this in another issue earlier:
https://github.com/louis-lau/DuckyAPI/issues/5#issuecomment-629251848

**Comment by louis-lau** [comment]
Closing this as I feel this has been resolved, let me know if it hasn't :)
